testjson/format.go

```diff
@@ -513,3 +526,85 @@ func githubActionsFormat(out io.Writer) EventFormatter {
 		return buf.Flush()
 	})
 }
+
+// writeGitHubActionsError parses test output and emits GitHub Actions error annotations
+func writeGitHubActionsError(
+	buf *bufio.Writer, event TestEvent, outputLines []string, fileLinePattern, panicStackPattern *regexp.Regexp,
+) {
+	sanitize := func(s string) string {
+		// Percent must be escaped first
+		s = strings.ReplaceAll(s, "%", "%25")
+		// Escape newlines and carriage returns
+		s = strings.ReplaceAll(s, "\r", "%0D")
+		s = strings.ReplaceAll(s, "\n", "%0A")
+		return s
+	}
+
+	// Check if this is a panic by looking for panic: in the output
+	var isPanic bool
+	var panicMessage strings.Builder
+	for _, outputLine := range outputLines {
+		if strings.Contains(outputLine, "panic:") {
```

Comment:
I'm a bit worried this would be true for anything returning the string `panic:`

The next loop and if would use the regexp to extract the info

Reply:
Do you mean to do this process

> "panic:", loop, then test with regexp, and you match once, apply the panic/test results


For every time we see the `panic:` match or only the once?

What this does is to gather up all the messages that match `panic:` then attributes them to the matching stack trace in the next if/loop block. 

Happy to make the necessary change. Just want to make sure I'm writing it up for the correct scenario you're thinking about 😅